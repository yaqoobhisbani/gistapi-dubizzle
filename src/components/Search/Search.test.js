import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Search from "./Search";

describe("Search component", () => {
  test("renders the search input", () => {
    const { getByPlaceholderText } = render(<Search />);
    const inputElement = getByPlaceholderText("Search Gists for the username");
    expect(inputElement).toBeInTheDocument();
  });

  test("calls onChange handler when search input changes", () => {
    const mockOnChange = jest.fn();
    const { getByTestId } = render(
      <Search search="" onChange={mockOnChange} />
    );
    const inputElement = getByTestId("input");
    fireEvent.change(inputElement, { target: { value: "test" } });
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });

  test("matches snapshot", () => {
    const { container } = render(<Search />);
    expect(container).toMatchSnapshot();
  });
});
