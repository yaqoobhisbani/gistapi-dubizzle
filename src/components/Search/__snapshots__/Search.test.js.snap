// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches snapshot 1`] = `
<div>
  <div
    class="sc-beqWaB ihoHfr"
  >
    <div
      class="sc-gueYoa eOwMm"
    >
      <span
        class="octicon octicon-search"
      />
      <input
        class="sc-dmqHEX bqMem"
        data-testid="input"
        placeholder="Search Gists for the username"
        value=""
      />
    </div>
  </div>
</div>
`;
